### PR TITLE
Integrate Keystone

### PR DESCRIPTION
**Pull request by zhiying-fan**
### Description

This PR include two parts of work:
1. Add basic support for connect Keystone Wallet and get extended public key from Keystone Wallet.
2. Sign transaction using Keystone.

#### For adding owner key

- Add `OwnerInfoKeystoneFragment` to start the QR code scanner.
- Add `KeystoneOwnerSelectionFragment` to show public keys.
- Add `PublicAddressDerivator` to derive public keys.
- Add one more field on `Owner` named `sourceFingerprint` and migrated the database to version 8.

#### For signing

- Add `KeystoneRequestSignatureFragment` for the UI of signing, `KeystoneSignViewModel` to handle the logic.
- Support for signing message for signing modes:
  - `CONFIRMATION`
  - `REJECTION`
  - `INITIATE_TRANSFER`

The logic for signing other signing mode in the future is also implemented, which is included in `KeystoneSignViewModel`.

### Dependencies

- [keystone-sdk-android](https://github.com/KeystoneHQ/keystone-sdk-android): Our library used to get public key from UR and sign transaction.

### TBD

Tracking events name

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/test/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModelTest.kt
Comment (elgatovital) [inline app/src/test/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModelTest.kt:1]: to fix tests, please add on line 517
```
  val someAddress = "0x1230B3d59858296A31053C1b8562Ecf89A2f888b".asEthereumAddress()!!
  coEvery { safeRepository.getActiveSafe() } returns Safe(someAddress, "safe_name")
```

File: app/src/main/java/io/gnosis/safe/Tracker.kt [hunk 1]
```diff
@@ -198,6 +206,7 @@ class Tracker internal constructor(
         val KEY_GENERATED = "user_key_generated"
         val KEY_IMPORTED = "user_key_imported"
         val KEY_IMPORTED_LEDGER = "user_ledger_nano_x_key_imported"
+        val KEY_IMPORTED_KEYSTONE = "user_keystone_key_imported"
         val KEY_DELETED = "user_key_deleted"
         val TRANSACTION_CONFIRMED = "user_transaction_confirmed"
         val TRANSACTION_CONFIRMED_LEDGER = "user_transaction_confirmed_ledger_nano_x"
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/Tracker.kt:212]: please add 
```kotlin 
val TRANSACTION_CONFIRMED_KEYSTONE = "user_transaction_confirmed_keystone"
```
and
```kotlin 
val TRANSACTION_REJECTED_KEYSTONE = "user_transaction_rejected_keystone"
```
no need to wire these events for now. We will do it later ourselves.

**Comment by liliya-soroka** [comment]
Found issues:
1. https://github.com/safe-global/safe-android/issues/1916

**Comment by liliya-soroka** [comment]
2. https://github.com/safe-global/safe-android/issues/1922

**Comment by TanyaEfremova** [comment]
3. Please review this design file for the images on screen` screen_owner_keystone_info`  [Open Figma file](https://www.figma.com/file/nCXULmvLaY6zv0fhizslkh/Keystone-Wallet-Integration?type=design&node-id=1019%3A7150&t=uoWihWRmeboo3rXg-1)

**Comment by zhiying-fan** [comment]
> 3. Please review this design file for the images on screen` screen_owner_keystone_info`  [Open Figma file](https://www.figma.com/file/nCXULmvLaY6zv0fhizslkh/Keystone-Wallet-Integration?type=design&node-id=1019%3A7150&t=uoWihWRmeboo3rXg-1)

Updated to latest, please have a look again.

**Comment by liliya-soroka** [comment]
4. https://github.com/safe-global/safe-android/issues/1925
5. https://github.com/safe-global/safe-android/issues/1926

**Comment by zhiying-fan** [comment]
> @zhiying-fan , any progress for point 2(the issue with the signing QR code) ?

@liliya-soroka I'm looking at the issue, and the new issues as well.

**Comment by elgatovital** [comment]
@zhiying-fan could you please sign the CLA by commenting `I have read the CLA Document and I hereby sign the CLA` ?

**Comment by zhiying-fan** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by elgatovital** [comment]
recheckcla

**Review by elgatovital — APPROVED** [review]

**Pull request merged**